src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3217,8 +3239,18 @@ protected override void VisitArguments(ImmutableArray<BoundExpression> arguments
                         parameterAnnotations,
                         results[i],
                         invokedAsExtensionMethod && i == 0);
+
+                    if (notNullParametersBuilder is object && results[i].RValueType.IsNotNull && (method.Parameters.Last().Equals(parameter) || !expanded))
+                    {
+                        // Note: expanded params parameters don't count
+                        notNullParametersBuilder.Add(parameter);
+                    }
                 }
+            }
 
+            var notNullParameters = notNullParametersBuilder?.ToImmutableAndFree() ?? ImmutableArray<ParameterSymbol>.Empty;
```

Comment:
>notNullParametersBuilder?.ToImmutableAndFree() ?? ImmutableArray<ParameterSymbol>.Empty [](start = 36, length = 87)

Consider leaving as `ArrayBuilder<ParameterSymbol>` rather than allocating an array, assuming the `ArrayBuilder` is not returned from this method.